### PR TITLE
[codein] Improve `CsvFileInput` performance using memory-mapped file

### LINKED ISSUE #33: Improve `CsvFileInput` performance using memory-mapped file

### PR DESCRIPTION
**Pull request by codein-dev**
Fixes #33

### CODE REVIEW

File: src/mongo/db/storage/csv_file.cpp [hunk 1]
```diff
@@ -29,10 +29,17 @@
...
+#include <fcntl.h>
+#include <filesystem>
 #include <fmt/format.h>
 #include <fstream>  // IWYU pragma: keep
 #include <memory>
 #include <string>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
```
Comment (MaxJuneKim) [inline src/mongo/db/storage/csv_file.cpp]: Why do we need sys/stats.h , sys/types.h, and unistd.h, and fcnti.h?
I get we need charcov for from_char(),
filesystem for filesystem, and sys/mman.h for mmap?

File: src/mongo/db/storage/csv_file.h [hunk 2]
```diff
@@ -99,18 +101,23 @@ class CsvFileInput : public StreamableInput {
...
     // metadata. It will return boost::none if there is no more line to read in the csv file.
     boost::optional<BSONObj> readBsonObj();
 
     std::string _fileAbsolutePath;
     std::string _metadataAbsolutePath;
-    std::ifstream _ifs;
+    int _fd = -1;
+    size_t _fileSize = 0;
+    const char* _data = nullptr;
+    size_t _offset = 0;
```
Comment (MaxJuneKim) [inline src/mongo/db/storage/csv_file.h]: What exactly is _fd for?
I figure that _fileSize is the size of the csv file,
_data is the mapped memory that holds content of the file in RAM, and _offset is the cursor that represents where we are reading in the file?

**Review by MaxJuneKim — APPROVED** [review]
I suppose this addressed issue #19 as well?
LGTM

File: src/mongo/db/storage/csv_file.cpp (hunk 1 above)
Comment (codein-dev) [reply src/mongo/db/storage/csv_file.cpp]: As we discussed offline, stats.h & types.h are not necessary anymore. They were just leftovers from the previous iteration. unistd.h is necessary to use `open()` and fcntl.h is necessary to use `O_RDONLY`.

File: src/mongo/db/storage/csv_file.h (hunk 2 above)
Comment (codein-dev) [reply src/mongo/db/storage/csv_file.h]: `fd` is a common acronym for a file descriptor.

**Review by codein-dev — COMMENTED** [review]
Thanks for review! Answered your comments

**Comment by codein-dev** [comment]
> I suppose this addressed issue #19 as well? LGTM

Yes, it did.

**Pull request merged**